File: frontend/src/interface/Customer.ts
// src/interface/Customer.ts
import type dayjs from "dayjs";

export interface Customer {
    ID: number; // เปลี่ยนจาก string เป็น number
    first_name: string;
    last_name: string;
    Password: string;
    Email: string;
    phone: string;
    Birthday: dayjs.Dayjs;
}

File: backend/entity/Customer.go
package entity

import (
	"gorm.io/gorm"
)

type Customer struct {
	gorm.Model
	Password  string `json:"password"`
	Email     string `json:"email"`
	Phone     string `json:"phone"`
	FirstName string `json:"first_name"`
	LastName  string `json:"last_name"`
	Birthday  string `json:"birthday"`

	PickupDelivery         []PickupDelivery        `gorm:"foreignKey:CustomerID" json:"pickup_deliveries"`
	InspectionAppointments []InspectionAppointment `gorm:"foreignKey:CustomerID" json:"inspection_appointments"`
	SalesContract          []SalesContract         `gorm:"foreignKey:CustomerID" json:"sales_contracts"`

	Payment []Payment `gorm:"foreignKey:CustomerID" json:"payments"`
}
